Set correct shared library name.

Override the arch setting to remove OpenBSD version from it,
so ports don't have to be bumped when OpenBSD version changes.

Index: configure
--- configure.orig
+++ configure
@@ -10999,7 +10999,7 @@ esac
     case "$target_cpu" in #(
   aarch64) :
 
-	for opt in -mbranch-protection=pac-ret -msign-return-address=all
+	for opt in -mbranch-protection=standard -msign-return-address=all
 do :
 
 
@@ -32012,7 +32012,7 @@ fi
 	 ;; #(
   openbsd*|mirbsd*) :
 
-	LIBRUBY_SO='lib$(RUBY_SO_NAME).$(SOEXT).$(MAJOR).'`expr ${MINOR} \* 10 + ${TEENY}`
+	LIBRUBY_SO='lib$(RUBY_SO_NAME).so.'${LIBruby33_VERSION}
 	 ;; #(
   solaris*) :
 
@@ -34138,7 +34138,7 @@ then :
 
 else $as_nop
 
-        arch="${target_cpu}-${target_os}"
+        arch="${target_cpu}-openbsd"
 
 fi
     printf "%s\n" "#define RUBY_PLATFORM \"$arch\"" >>confdefs.h
